### PR TITLE
Move dependencies into pyproject

### PR DESCRIPTION
**Pull request by pgunn**
Addresses #1417 

Thanks to @sneakers-the-rat for prodding us on this.

I made jupyter the optional dependency here; we could also break the "dev" tools out, but the test functionality of caimanmanager depends on mypy and I'd rather not go through the fuss to make it optional (mypy is a fairly small and no-fuss dependency anyhow, so far).

### CODE REVIEW

**Comment by pgunn** [comment]
Seems to pass at least preliminary testing using a bare conda environment and a pip-only install. It's not a very fast install route, but it seems to work.

File: pyproject.toml [hunk 1]
```diff
@@ -11,6 +11,43 @@ authors = [
...
+  "tensorflow",
+  "tifffile",
+  "tqdm",
+  "yapf",
+  "zarr"
+]
+
+[project.optional-dependencies]
+jupyter = [
+  "jupyter",
```
Comment (kushalkolar) [inline pyproject.toml:47]: It doesn't make sense for jupyter to be optional but bokeh mandatory?

Comment (pgunn) [reply pyproject.toml:47]: CNMF and visualization both import bokeh; I think we need it to be present until some future cleanup changes that.

I don't think bokeh pulls jupyter in.

Comment (kushalkolar) [reply pyproject.toml:47]: Ah right. Though none of the bokeh viz will actually work without jupyter AFAIK.

**Pull request merged**